torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -152,36 +152,40 @@ IterDomain* TensorView::axis(int pos) const {
   return domain()->axis(pos);
 }
 
-void TensorView::setComputeAt(TensorView* computeAtView, int axis) {
-  compute_at_view_ = computeAtView;
-  relative_compute_at_axis_ = axis;
-  setThisComputeAtAxis();
-
-  TORCH_INTERNAL_ASSERT(
-      getThisComputeAtAxis() >= 0 &&
-          (unsigned int)getThisComputeAtAxis() <= nDims(),
-      "Invalid computeAt on ",
-      this,
-      " tried to set to local axis ",
-      getThisComputeAtAxis());
-
-  TORCH_INTERNAL_ASSERT(
-      std::none_of(
-          domain()->domain().begin(),
-          domain()->domain().begin() + getThisComputeAtAxis(),
-          [](IterDomain* id) { return id->isReduction(); }),
-      "Invalid computeAt, reduction domain inside computeAt axis.");
-}
-
 void TensorView::setComputeAt(
     TensorView* computeAtView,
     int thisPos,
     int relPos) {
+  TORCH_INTERNAL_ASSERT(
+      thisPos > 0 && (unsigned)thisPos <= nDims(),
+      "Invalid this computeAt position for T",
+      name(),
+      ": ",
+      thisPos);
+  // When computeAtView is a consumer, the CA axes must not include
+  // reductions. Note that an output tensor may be set as computed at
+  // another output tensor even if they are not a producer and a
+  // consumer.
+  if (isConsumer(computeAtView)) {
```

Comment:
Do you want to validate else both are outputs?

Reply:
That's a good idea. Validation added.